contrib/babelfishpg_tsql/src/hooks.c

```diff
@@ -2689,6 +2699,169 @@ pltsql_detect_numeric_overflow(int weight, int dscale, int first_block, int nume
 	return (total_digit_count > TDS_NUMERIC_MAX_PRECISION);
 }
 
+/*
+ * Throw a common error message while casting to time datatype
+ */
+static void gen_func_time_in_error()
+{
+	ereport(ERROR,
+			(errcode(ERRCODE_INVALID_DATETIME_FORMAT),
+			errmsg("Conversion failed when converting date and/or time from character string.")));
+}
+
+static bool containsInMonthFormat(int *ftype){
```

Comment:
Will add an explanation that this function will check if the first 3 inputs are of format yyyy mm dd then will send true so that we can convert the input to yyyy-mm-dd